### PR TITLE
Cleaner PIMPL idiom for ImageBuf

### PR DESCRIPTION
**Pull request by lgritz**
This is a revision of #2350. I like this solution better.

### CODE REVIEW

File: src/include/OpenImageIO/imagebuf.h [hunk 1]
```diff
@@ -1408,13 +1401,12 @@ class OIIO_API ImageBuf {
...
+    // PIMPL idiom
+    static void impl_deleter(ImageBufImpl*);
+    std::unique_ptr<ImageBufImpl, decltype(&impl_deleter)> m_impl;
 
-    ImageBufImpl* impl() { return (ImageBufImpl*)m_impl.get(); }
-    const ImageBufImpl* impl() const
-    {
-        return (const ImageBufImpl*)m_impl.get();
-    }
+    ImageBufImpl* impl() { return m_impl.get(); }
```
Comment (fpsunflower) [inline src/include/OpenImageIO/imagebuf.h]: Does this help anything? Anything that can access `impl()` can also access `m_impl` which is the same length.

I doesn't look like `impl()` is used too many times in the code either ...

**Comment by fpsunflower** [comment]
I like this cleanup!

File: src/include/OpenImageIO/imagebuf.h (hunk 1 above)
Comment (lgritz) [reply src/include/OpenImageIO/imagebuf.h]: Oh, you might be right. I'll take a look and eliminate it if it is only in a few places.

Comment (lgritz) [reply src/include/OpenImageIO/imagebuf.h]: It was a LOT of places. I just pushed an update, see if you are comfortable with that much change.

Comment (fpsunflower) [reply src/include/OpenImageIO/imagebuf.h]: I like it, I think this will be cleaner in the long run.

**Comment by fpsunflower** [comment]
LGTM!

**Pull request merged**